providers/google.go

```diff
@@ -98,17 +100,13 @@ func NewGoogleProvider(p *ProviderData, opts options.GoogleOptions) (*GoogleProv
 		},
 	}
 
-	if opts.ServiceAccountJSON != "" {
-		file, err := os.Open(opts.ServiceAccountJSON)
-		if err != nil {
-			return nil, fmt.Errorf("invalid Google credentials file: %s", opts.ServiceAccountJSON)
-		}
-
+	if opts.ServiceAccountJSON != "" || opts.UseApplicationDefaultCredentials {
```

Comment:
Had to reintroduce this if statement to make the test working again, which is also valid since the Groups feature will only work with a Service Account (file or ADC) set.